test/e2e/upgrades/upgrade.go

```diff
@@ -41,6 +41,9 @@ const (
 
 // Test is an interface for upgrade tests.
 type Test interface {
+	// Name should return a test name sans spaces.
+	Name() string
```

Comment:
Yea. This really is just moving the burden on the test writer to remember to use their own namespace by calling `framework.CreateNamespace` to just implement this method. And this one is a compile time check. As you can see, a couple of tests got by review (probably by me) without remembering to do this.